File: 100-Days-Series/Day 57/NumberPyramid.js

// Arrow Function to print Pyramid
const getPrintNumberPyramid=(n)=>{
    for(let i=1;i<=n;i++){
        line="";
        // collect spaces
        for(let j=i;j<n;j++){
            line+=" ";
        }
        // collect ascending numbers
        for(let j=1;j<=i;j++){
            line+=j;
        }
        // collect descending numbers
        for(let j=i-1;j>=1;j--){
            line+=j;
        }
        console.log(line);
    }
}
// Let's Start here
rows=5;
getPrintNumberPyramid(rows);

File: 100-Days-Series/Day 57/NumberPyramid.py

# Function to print Pyramid
def getPrintNumberPyramid(n):
    for i in range(1,n+1,1):
        #Print spaces
        for j in range(i,n,1):
            print(" ",end="")
        #Print Ascending numbers
        for j in range(1,i+1,1):
            print(j,end="")
        #Print Descending numbers
        for j in range(i-1,0,-1):
            print(j,end="")
        print()

# Let's Start here
rows=int(input("Enter the number of rows : "))
getPrintNumberPyramid(rows)